File: 2163-kth-distinct-string-in-an-array/kth-distinct-string-in-an-array.java
class Solution {
    public String kthDistinct(String[] arr, int k) {
        HashMap<String, Integer> map = new HashMap<>();
        for(String s: arr){
            if(map.containsKey(s))map.put(s, map.get(s)+1);
            else map.put(s, 1);
        }
        int x=  0, i = 0;
        while(i < arr.length){
            if(map.get(arr[i]) == 1)x++;
            if(x == k)return arr[i];
            i++;
        }
        return "";
    }
}

File: 2163-kth-distinct-string-in-an-array/kth-distinct-string-in-an-array.cpp
#include<bits/stdc++.h>
class Solution {
public:
    string kthDistinct(vector<string>& arr, int k) {
        unordered_map<string,int>mp;
        for(auto it:arr){
            mp[it]++;
        }
        vector<pair<int,string>>ans;
        for(auto it:mp){
            if(it.second==1){
                for(int i=0;i<arr.size();i++){
                    if(it.first==arr[i]){
                        ans.push_back(make_pair(i,it.first));
                    }
                }
            }
        }
        if(ans.size()<k){
            return "";
        }
        sort(ans.begin(),ans.end());
        return ans[k-1].second;
    }
};